src/Extension/Publishable/PublishableSiteTree.php

```diff
@@ -87,9 +88,13 @@ public function objectsToDelete($context)
 
     /**
      * The only URL belonging to this object is it's own URL.
+     * Instead of Link() which calls VersionedStateExtension::updateLink and appends stage=Live
      */
     public function urlsToCache()
     {
-        return [Director::absoluteURL($this->getOwner()->Link()) => 0];
+        $relativeLink = $this->getOwner()->RelativeLink();
+        $link =  Controller::join_links(Director::baseURL(), $relativeLink);
+
+        return [Director::absoluteURL($link) => 0];
```

Comment:
```suggestion
        return Versioned::withVersionedMode(function () {
            Versioned::set_reading_mode(Versioned::get_default_reading_mode());
            return [Director::absoluteURL($this->getOwner()->Link()) => 0];
        });
```

Reply:
Something like this?

Reply:
Although I know that query string are not supported, but the fix does not have to know whether there is versioning or not.
In this example, it clearly couples with versioning in mind.

Let me try the code and see if it will still work as expected.

Reply:
I still get `[2021-05-21 21:44:56] error-log.ALERT: Fatal Error (E_ERROR): Maximum execution time of 30 seconds exceeded {"code":1,"message":"Maximum execution time of 30 seconds exceeded","file":"/var/www/staticpublish/vendor/silverstripe/framework/src/Core/Injector/InjectionCreator.php","line":17,"trace":null} []
`

Reply:
@dhensby It still appends `?stage=Live` to all pages as shown in my previous screenshot which I think versioned still knows that it is still in Live mode.

Another problem I see is that if there is a query string like that, then `Publisher::generatePageResponse($url)` is going to redirect to login page since this is coming from a different session.

Reply:
Hey @ssmarco , can you please check what user is used when running your job? You can check this by looking at `RunAsMemberID` DB column. I think we had a similar issue on some other project. Static cache jobs should run without a user to force correct stage (`RunAsMemberID` = 0).

Reply:
Hi @mfendeksilverstripe, I saw `RunAsID` in `QueuedJobDescriptor` table and all values are zero

Reply:
That's the correct value so this is indeed an issue that needs to be fixed. I had a look at one of the projects where we use this module and this is the current solution:

```
// We never want to have stage=Stage present in our cached URLs. This will safely remove "stage" params, but
// keep any others. It doesn't matter where in the string "stage=" exists.
$url = preg_replace('/([?&])stage=[^&]+(&|$)/', '$1', $url);
// Trim any trailing "?" or "&".
$url = $string = rtrim($url, '&');
$url = $string = rtrim($url, '?');
```
I added some [extensibility improvements](https://github.com/silverstripe/silverstripe-staticpublishqueue/pull/132/files#diff-6552cf24d9678ff8dfa09b29e7d6b2437c951075d96ffd6dbe558e827234f52dR17) in one of my PRs so maybe fix for this can be included in the URLService..